hashchain.c

```diff
@@ -76,6 +76,9 @@ bool hash_chain_verify(const void *h,
     EVP_MD_CTX *ctx;
     int result;
     int digest_len = EVP_MD_size(hash);
+    if (memcmp(h, tip, digest_len) == 0)
+        return 1;
```

Comment:
By the way, it should be `true` or `false` since its return type is specific to `bool`. Please send another pull request to rework.

Reply:
Fixed.